File: spec/python/test_nav_root.py
import unittest

from nav_root import NavRoot

class TestNavRoot(unittest.TestCase):
    def test_nav_root(self):
        r = NavRoot.from_file("src/nav.bin")

        self.assertEqual(r.header.qty_entries, 2)
        self.assertEqual(r.header.filename_len, 8)

        self.assertEqual(len(r.index.entries), 2)
        self.assertEqual(r.index.entries[0].filename, "FIRST___")
        self.assertEqual(r.index.entries[1].filename, "SECOND__")

File: spec/java/src/io/kaitai/struct/spec/TestNavRoot.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.*;
import org.testng.annotations.Test;

import java.nio.charset.Charset;
import java.util.ArrayList;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertEqualsNoOrder;

public class TestNavRoot extends CommonSpec {
    @Test
    public void testNavRoot() throws Exception {
        NavRoot r = NavRoot.fromFile(SRC_DIR + "nav.bin");

        assertEquals(r.header().qtyEntries(), 2);
        assertEquals(r.header().filenameLen(), 8);

        assertEquals(r.index().entries().size(), 2);
        assertEquals(r.index().entries().get(0).filename(), "FIRST___");
        assertEquals(r.index().entries().get(1).filename(), "SECOND__");
    }

}
